omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/AdministrationLinksResource1_8.java

```diff
@@ -0,0 +1,183 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_8;
+
+import io.swagger.models.Model;
+import io.swagger.models.ModelImpl;
+import io.swagger.models.properties.ArrayProperty;
+import io.swagger.models.properties.ObjectProperty;
+import io.swagger.models.properties.StringProperty;
+import org.openmrs.module.Extension;
+import org.openmrs.module.web.extension.AdministrationSectionExt;
+import org.openmrs.module.webservices.helper.ModuleFactoryWrapper;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.PropertyGetter;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.RefRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingReadableResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ObjectNotFoundException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.openmrs.module.webservices.rest.web.v1_0.wrapper.openmrs1_8.ModuleAdministrationLinks1_8;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Resource(name = RestConstants.VERSION_1 + "/administrationlinks", supportedClass = ModuleAdministrationLinks1_8.class,
+		supportedOpenmrsVersions = { "1.8.*", "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*", "2.1.*", "2.2.*", "2.3.*",
+				"2.4.*", "2.5.*" })
+public class AdministrationLinksResource1_8 extends BaseDelegatingReadableResource<ModuleAdministrationLinks1_8> {
+
+	private static final String ADMIN_LIST_POINT_ID = "org.openmrs.admin.list";
+
+	private static final String MODULE_TITLE = "moduleTitle";
+
+	private static final String LINKS = "administrationLinks";
+
+	/**
+	 * ModuleFactoryWrapper is used for testing purposes.
```

Comment:
If used for only testing purposes, then why include in an runtime class? Shouldn't it be under test resources?

Reply:
It's a wrapper for ModuleFactory that I can then mock in tests. There is no other way to mock ModuleFactory in this class if I'm not using a ModuleFactoryWrapper

Reply:
The same technique is used in ModuleActionResource1_8